src/js/form.js

```diff
@@ -413,7 +413,10 @@ Form.prototype.replaceChoiceNameFn = function( expr, resTypeStr, context, index,
             let label = '';
             const value = this.model.evaluate( params[ 0 ], resTypeStr, context, index, tryNative );
             const name = stripQuotes( params[ 1 ] ).trim();
-            const inputs = [ ...this.view.html.querySelectorAll( `[name="${name.startsWith( '/' ) ? name : joinPath( context, name )}"]` ) ];
+            var inputs = [ ...this.view.html.querySelectorAll( `[name="${name.startsWith( '/' ) ? name : joinPath( context, name )}"]` ) ];
+            inputs = inputs.length ?
+                    inputs :
+                    [ ...this.view.html.querySelectorAll( `[data-name="${name.startsWith( '/' ) ? name : joinPath( context, name )}"]` ) ];
```

Comment:
Nice work!

I'd slightly prefer searching directly for `[name="${name}"],[data-name="${name}"]` in one go (and update the name variable beforehand).

Some background:

Only radio buttons don't use the path in the `name` attribute. The reason is that this would make all repeated questions with radiobuttons behave as one. Hence, we use the unfortunate `data-name` attribute for radiobuttons. In hindsight, it would have been better to not use name at all but use data-name (or better: data-path) for everything.